examples/Canvas/main.lua

```diff
@@ -0,0 +1,148 @@
+local screenShaderVertex = [[
+  vec4 position(mat4 projection, mat4 transform, vec4 vertex) {
+    return vertex;
+  }
+]]
+
+-- For the fragment shader: We are going to create a separable gaussian blur.
+-- A "separable" blur means we first blur horizontally, then blur vertically to get a 2D blur.
+local screenShaderFragment = [[
+  // This one-dimensional blur filter samples five points and averages them by different amounts.
+  // Weights and offsets taken from http://rastergrid.com/blog/2010/09/efficient-gaussian-blur-with-linear-sampling/
+
+  // The weights for the center, one-point-out, and two-point-out samples
+  #define WEIGHT0 0.2270270270
+  #define WEIGHT1 0.3162162162
+  #define WEIGHT2 0.0702702703
+
+  // The distances-from-center for the samples
+  #define OFFSET1 1.3846153846
+  #define OFFSET2 3.2307692308
+
+  // UVs are sampled from a texture over the range 0..1.
+  // This uniform is set outside the shader so we know what UV distance "one pixel" is.  
+  uniform vec2 resolution;
+  
+  // This uniform will be set every draw to determine whether we are sampling horizontally or vertically.
+  uniform vec2 direction;
+  
+  // lovr's shader architecture will automatically supply a main(), which will call this color() function 
+  vec4 color(vec4 graphicsColor, sampler2D image, vec2 uv) {
+    vec2 pixelOff = direction / resolution;
+    vec4 color = vec4(0.0);
+    color += texture(image, uv) * WEIGHT0;
+    color += texture(image, uv + pixelOff * OFFSET1) * WEIGHT1;
+    color += texture(image, uv - pixelOff * OFFSET1) * WEIGHT1;
+    color += texture(image, uv + pixelOff * OFFSET2) * WEIGHT2;
+    color += texture(image, uv - pixelOff * OFFSET2) * WEIGHT2;
+    return color;
+  }
+]]
+
+-- The vertex and fragment shaders will be combined together into a shader program
+local screenShader
+
+-- Image of an eyechart
+local eyechart
+
+-- This table will contain two canvases we will use as scratch space
+local tempCanvas
+
+function lovr.load()
+  -- Load the eyechart image
+  -- Source: https://www.publicdomainpictures.net/view-image.php?image=244244&picture=eye-chart-test-vintage
+  -- Creative Commons 0 / Public Domain license
+  local texture = lovr.graphics.newTexture('eye-chart-test-vintage-cc0.jpg')
+  local textureWidth, textureHeight = texture:getDimensions()
+  eyechart = {
+    aspect = textureHeight / textureWidth,
+    material = lovr.graphics.newMaterial( texture )
+  }
+
+  -- Configure the shader
+  if useCanvas then
+    local width, height = lovr.headset.getDisplayDimensions()
+    
+    -- Compile the shader
+    screenShader = lovr.graphics.newShader(screenShaderVertex, screenShaderFragment)
+
+    -- Set the resolution uniform
+    screenShader:send("resolution", {width, height})
+
+    -- Create two temporary canvases
+    tempCanvas = {
+      lovr.graphics.newCanvas(width, height),
+      lovr.graphics.newCanvas(width, height)
+    }

```

Comment:
Can you add something along the lines of `tempCanvas[]:setWrap('clamp')` so the blur doesn't bleed along the edges?